metecho/api/models.py

```diff
@@ -265,6 +265,7 @@ class Repository(
         validators=[validate_unicode_branch],
         default="master",
     )
+    branch_prefix = StringField(blank=True)
```

Comment:
I see many places we're using `StringField` with `null=True, blank=True`, but other places just `blank=True` -- I don't fully appreciate the distinction.

Reply:
Any character-type field (`StringField`, `CharField`, `TextField`, etc) can hold `None` if it is `null=True`, thus making its type `Optional[str]`, or can hold the empty string if it is `blank=True`, thus (assuming it is not also `null=True`) keeping its type `str`.

Reply:
Ok, I think I follow that. Is there a reason we're using it both ways? Is there one we prefer?

Reply:
The way I think of it is that `null` controls whether the field can store None or not, and `blank` controls whether the Django admin UI (and I guess DRF also?) will allow entering an empty string.

Reply:
I was under the impression that `null=True` was generally discouraged on text fields?

> Avoid using null on string-based fields such as CharField and TextField.

https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.Field.null